File: backend/app/schemas/evaluation.py
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field


EvaluationStatus = Literal[
    "queued",
    "processing",
    "completed",
    "failed",
]


CriterionResultStatus = Literal[
    "satisfied",
    "not_satisfied",
    "uncertain",
    "not_applicable",
    "manual_review",
    "not_evaluated",
    "failed",
]


class StartEvaluationRequest(BaseModel):
    criterion_set_id: str


class EvaluationSummary(BaseModel):
    id: str
    project_id: str
    criterion_set_id: str

    status: EvaluationStatus

    total_score: float
    maximum_score: float
    score_percentage: float

    satisfied_count: int
    not_satisfied_count: int
    uncertain_count: int
    manual_review_count: int

    started_at: datetime | None
    completed_at: datetime | None

    error_message: str | None
    created_at: datetime


class CriterionEvaluationResult(BaseModel):
    id: str
    criterion_id: str

    criterion_code: str
    criterion_title: str
    category_name: str | None

    result_status: CriterionResultStatus

    awarded_score: float
    maximum_score: float

    extracted_values: dict[str, Any]
    citations: list[dict[str, Any]]

    evidence_summary: str | None

    missing_information: list[str]
    warnings: list[str]

    confidence: float
    requires_manual_review: bool


class EvaluationDetail(BaseModel):
    evaluation: EvaluationSummary
    results: list[CriterionEvaluationResult]

File: frontend/types/evaluation.ts
export type EvaluationStatus =
  | "queued"
  | "processing"
  | "completed"
  | "failed";

export type CriterionResultStatus =
  | "satisfied"
  | "not_satisfied"
  | "uncertain"
  | "not_applicable"
  | "manual_review"
  | "not_evaluated"
  | "failed";

export interface CriterionSet {
  id: string;
  code: string;
  name: string;
  version: string;
  description: string | null;
  status: string;
}

export interface EvaluationSummary {
  id: string;
  project_id: string;
  criterion_set_id: string;

  status: EvaluationStatus;

  total_score: number;
  maximum_score: number;
  score_percentage: number;

  satisfied_count: number;
  not_satisfied_count: number;
  uncertain_count: number;
  manual_review_count: number;

  started_at: string | null;
  completed_at: string | null;
  error_message: string | null;

  created_at: string;
}

export interface CriterionResult {
  id: string;

  result_status: CriterionResultStatus;

  awarded_score: number;
  maximum_score: number;

  extracted_values: Record<
    string,
    unknown
  >;

  citations: Array<{
    source_id: string;
    document_id: string;
    original_filename: string;
    page_number: number;
    similarity: number;
    excerpt: string;
  }>;

  evidence_summary: string | null;

  missing_information: string[];
  warnings: string[];

  confidence: number;
  requires_manual_review: boolean;

  criteria: {
    code: string;
    title: string;
    category_code: string | null;
    category_name: string | null;
    display_order: number;
  };
}

export interface EvaluationDetail {
  evaluation: EvaluationSummary;
  results: CriterionResult[];
}